hollow/src/main/java/com/netflix/hollow/api/consumer/incubating/HollowConsumerV2.java

```diff
@@ -0,0 +1,515 @@
+package com.netflix.hollow.api.consumer.incubating;
+
+import static java.lang.System.arraycopy;
+import static java.util.Collections.emptyList;
+import static java.util.Collections.emptySet;
+import static java.util.Objects.requireNonNull;
+import static java.util.stream.Collectors.toList;
+
+import com.netflix.hollow.api.client.HollowAPIFactory;
+import com.netflix.hollow.api.consumer.HollowConsumer;
+import com.netflix.hollow.api.consumer.fs.HollowFilesystemBlobRetriever;
+import com.netflix.hollow.api.custom.HollowAPI;
+import com.netflix.hollow.api.metrics.HollowConsumerMetrics;
+import com.netflix.hollow.api.metrics.HollowMetricsCollector;
+import com.netflix.hollow.core.read.filter.HollowFilterConfig;
+import com.netflix.hollow.core.util.DefaultHashCodeFinder;
+import com.netflix.hollow.core.util.HollowObjectHashCodeFinder;
+import java.io.File;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.LinkedHashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.Executor;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.locks.ReadWriteLock;
+import java.util.concurrent.locks.ReentrantReadWriteLock;
+import java.util.stream.Collectors;
+
+public class HollowConsumerV2<A extends HollowAPI> extends HollowConsumer {
```

Comment:
Having parameterized `HollowConsumer` and `HollowConsumer.Builder` was one of the original motivations.

`HollowConsumer` is unparameterized currently, so this would be a backwards compatible change if it were done in `HollowConsumer`.